src/main/java/de/caritas/cob/userservice/api/service/emailsupplier/TenantDataSupplier.java

```diff
@@ -0,0 +1,53 @@
+package de.caritas.cob.userservice.api.service.emailsupplier;
+
+import de.caritas.cob.userservice.api.tenant.TenantContext;
+import de.caritas.cob.userservice.mailservice.generated.web.model.TemplateDataDTO;
+import de.caritas.cob.userservice.tenantservice.generated.web.TenantControllerApi;
+import de.caritas.cob.userservice.tenantservice.generated.web.model.RestrictedTenantDTO;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class TenantDataSupplier {
+
+  private final TenantControllerApi tenantControllerApi;
+
+  @Value("${app.base.url}")
+  private String applicationBaseUrl;
+
+  public List<TemplateDataDTO> getTemplateAttributes() {
+    var subdomain = TenantContext.getCurrentTenantData().getSubdomain();
```

Comment:
please split this method to smaller chunks with single responsibility and meaningful names to reduce complexity

Reply:
done